api/control/v1alpha1/types.go

```diff
@@ -443,6 +465,27 @@ const (
 // Type checking errors are posted to the status.Errors of the Binding and BindingPolicy.
 type Expression string
 
+// Augmentation defines what is implicitly added to every workload object from a WEC,
+// for purposes of status collection.
+type Augmentation struct {
+	Inventory   InventoryObjectReference `json:"inventory"`
+	Propagation PropagationMeta          `json:"propagation"`
+}
+
+// InventoryObjectReference identifies an inventory object.
+// Inventory is queryable in StatusCollector expressions using the `inventory` object.
+type InventoryObjectReference struct {
+	// `name` is the inventory object's name
+	Name string `json:"name"`
+}
+
+// PropagationMeta is the last reported metadata about one workload object at one WEC.
+// PropagationMeta is queryable in StatusCollector expressions using the `propagation` object.
+type PropagationMeta struct {
+	// `stale` indicates whether `infoAgeSeconds > stalenessThresholdSecs`.
```

Comment:
"infoAgeSeconds" is not defined.

Reply:
Added support for infoAgeSeconds.

Reply:
infoAgeSeconds is no longer defined.
LastUpdateTimestamp is not suitable for judging whether the core has stale information. The core might have very fresh information that the object was last updated a long time ago.

Reply:
Right. This can only be propagated by adding a new field to the WorkStatus API object and updating the ocm status agent.

Reply:
@MikeSpreitzer on 2nd thought, I think both may be interesting to the user.

It depends on how staleness is defined in this case. Staleness under the context of propagation does make sense when set by how recent the sync was. Staleness of objects on the WEC may be better suited to be a part of the `returned` object.

Reply:
Oh crud, I see that OCM's ManifestWork does not report on how fresh is the information in the core. It only reports on the time of the last change, not how recently the core was informed that there has been no further change. The point that I was trying to get at with "infoAge" is how old or fresh is the core's information from the WEC.

For this PR we have to content ourselves with something that is supported by ManifestWork. For a given workload object, I think the best that we can do is expose the binary condition of whether the version of that object that is currently in the WDS has been successfully propagated into the WEC. I think that would be derived from the conjunction of the following tests.

- Is the ResourceVersion read from the WDS equal to the ResourceVersion in the ManifestWork?
- Does the ObservedGeneration in the ManifestWork.Status.Conditions equal the Generation of the ManifestWork?
- Does the ManifestWork.Status.ResourceStatus.Manifests[the index for the object].Conditions have an entry for "Applied" with a status of "True"?
- Maybe also the same for "Available"? I am not sure exactly what this one means in the Conditions for an individual object (the comment in https://github.com/open-cluster-management-io/api/blob/main/work/v1/types.go looks like it was written only with ManifestWork.Status.Conditions in mind).